astropy/units/tests/test_format.py

```diff
@@ -315,6 +315,26 @@ def test_ogip_invalid_multiplication(string, message, unit):
         assert u_format.OGIP.parse(string) == unit
 
 
+@pytest.mark.parametrize(
+    "string,unit,power",
+    [
+        pytest.param("s**-1", u.s**-1, "-1", id="int_unit_power"),
+        pytest.param("m**-2.0", u.m**-2, "-2.0", id="float_unit_power"),
+        pytest.param("10**-3 kg", u.g, "-3", id="int_scale_power"),
+    ],
+)
+def test_ogip_negative_exponent_parenthesis(string, unit, power):
+    # Regression test for #16788 - negative powers require parenthesis
+    with pytest.warns(
+        UnitParserWarning,
+        match=(
+            r"^negative exponents must be enclosed in parenthesis\. "
+            rf"Expected '\*\*\({power}\)' instead of '\*\*{power}'\.$"
+        ),
+    ):
+        assert u_format.OGIP.parse(string) == unit
```

Comment:
Might it be an idea to also test the more usual public API in `test_units`, maybe adding to `test_parse_string_non_critical_error` - obviously, there it can just be a single example.

Reply:
The purpose of `test_parse_string_non_critical_error()` is not to test the parser rules. Its purpose is to test that `Unit()` obeys `parse_strict`.

Reply:
OK, let's just rely on the warning but still get OK unit already being tested sufficiently elsewhere.